browser/brave_shields/ad_block_only_mode_browsertest.cc

```diff
@@ -0,0 +1,230 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "base/functional/callback.h"
+#include "base/memory/scoped_refptr.h"
+#include "base/run_loop.h"
+#include "brave/components/brave_shields/core/browser/brave_shields_utils.h"
+#include "brave/components/brave_shields/core/common/features.h"
+#include "brave/components/brave_shields/core/common/pref_names.h"
+#include "brave/components/de_amp/common/pref_names.h"
+#include "brave/components/debounce/core/common/pref_names.h"
+#include "brave/components/global_privacy_control/pref_names.h"
+#include "brave/components/query_filter/pref_names.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/content_settings/cookie_settings_factory.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/test/base/chrome_test_utils.h"
+#include "chrome/test/base/platform_browser_test.h"
+#include "components/content_settings/core/browser/cookie_settings.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace brave_shields {
+
+class AdblockOnlyModeBrowserTestBase : public PlatformBrowserTest {
+ public:
+  AdblockOnlyModeBrowserTestBase() = default;
+  ~AdblockOnlyModeBrowserTestBase() override = default;
+
+  PrefService* local_state() { return g_browser_process->local_state(); }
+
+  void WaitForAdBlockOnlyModeChangeApplied() {
```

Comment:
This relies on that changes propagated via post task, do you mind to return a kind of waiter object here (e.g. `base::test::Future`) to register listener before you make changes. In this case it should work even if changes are handled synchronously